pkg/collect/run_pod.go

```diff
@@ -378,3 +391,37 @@ func RunPodLogs(ctx context.Context, client v1.CoreV1Interface, podSpec *corev1.
 
 	return ioutil.ReadAll(logs)
 }
+
+func savePodDetails(output CollectorResult, bundlePath string, clientConfig *rest.Config, pod *corev1.Pod, runPodCollector *troubleshootv1beta2.RunPod) (CollectorResult, error) {
+	client, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed create client from config")
+	}
+
+	ctx := context.Background()
+
+	podStatus, err := client.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get pod")
+	}
+
+	podEvents, err := client.CoreV1().Events(pod.Namespace).List(ctx, metav1.ListOptions{FieldSelector: (fmt.Sprintf("involvedObject.name=%s", pod.Name)), TypeMeta: metav1.TypeMeta{Kind: "Pod"}})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get pod events")
+	}
+
+	podBytes, err := json.MarshalIndent(podStatus, "", "  ")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal pod status")
+	}
+
+	podEventBytes, err := json.MarshalIndent(podEvents, "", "  ")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal pod events")
+	}
+
+	output.SaveResult(bundlePath, filepath.Join(runPodCollector.Name, fmt.Sprintf("%s.json", runPodCollector.CollectorName)), bytes.NewBuffer(podBytes))
```

Comment:
We ignore errors returned by these `SaveResult` quite often. I'd prefer we log these errors instead of "swallowing" them. I wonder whether we should have a `SaveResultLogError()` or `SaveResultSafe()` function that internally wraps `SaveResult` but logs any error and has no return value.

Reply:
Perhaps we should rename `SaveResult` to `SaveResultErr` to indicate it returns an error?